airflow/utils/sqlalchemy.py

```diff
@@ -101,6 +101,12 @@ def ping_connection(connection, branch):
     def connect(dbapi_connection, connection_record):
         connection_record.info['pid'] = os.getpid()
 
+    @event.listens_for(engine, "connect")
+    def set_sqlite_pragma(dbapi_connection, connection_record):
+        if 'sqlite3.Connection' in str(type(dbapi_connection)):
+            cursor = dbapi_connection.cursor()
+            cursor.execute("PRAGMA foreign_keys=ON")
```

Comment:
This is debatable. Without FK enabled, for SQLite, the migration script `2e82aab8ef20_rename_user_table.py` doesn't change the FK of `chart` and `known_event` tables, they would continue to point to `user` table which no longer exists. Alembic then fails with `NoSuchTableError`.

Drawback is that existing SQLite DBs are not fixed. My assumtion is that SQLite is not used in production environment and doesn't contain long-living data. If that assumption is wrong I can add an additional migration script that checks and fixes the wrong FK for SQLite.

Reply:
All the docs say "don't use SQLite for production", correct.

What will this do with when people run it against a local SQLite DB? Will it migrate without problems?  We want the test/"I'm playing with Airflow" experience to be nice too.

Reply:
I tested quickly, with master I created a user, a chart, and a known event. Then switched to this branch. Startup works, but with FK enabled it's no longer possible to create new known events or to edit links between user and event/chart. I'll try to create another migration script to fix it.

Reply:
Done, see 856955da8476_fix_sqlite_foreign_key.py
Same test, FK and data is now properly migrated.
One caveat: If the database already contains inconsistent data (I set an non-existing user_id) then migration will fail.